### ISSUE TITLE
Unity Crash on Startup

### ISSUE THREAD

**Issue by Xenoblivion53** [paulov-t/SIT.Core#60]
Attempted with Versions: | 3.5.7 BLEEDINGEDGE | SPT-AKI 3.5.6 | SPT-AKI 3.5.5 |
Used the latest release for the SIT Server Mod.

I'm barely even coherent with code, so there is a possibility I messed something up, but to explain, it is that when I start the server and launch the game via the SIT Launcher, it will get through the loading screen, but right before it "rotates" to the main menu buttons, Unity has an error, and the game crashes within seconds after.

I followed the Hamatchi guide and made sure I didn't miss anything involving the installation setup and file configurations.

I can't seem to find where the logs for this would be, so if they are needed, I would need some guidance on where to find them.

I look forward to being able to play with my friend, so any help would be greatly appreciated!

_Here are some basic screenshots in case I overlooked anything obvious:_

![image](https://github.com/paulov-t/SIT.Core/assets/82923762/be4f47ec-1cd7-4df6-8736-6e58beb50cec)

![image](https://github.com/paulov-t/SIT.Core/assets/82923762/d4bfcdb7-a6d3-43c1-bcb9-dd43744fe5fc)

![image](https://github.com/paulov-t/SIT.Core/assets/82923762/d2b74afb-9bf3-4f8d-b5d5-ca84ff932e01)

**Sorry if I seem a little dense about this, I'm clueless why this is happening and have given up trying to fix it myself.**

**Comment by w123456w30w** [comment]
make sure you have latest version game file

**Comment by Xenoblivion53** [comment]
Thank you! Updating Tarkov and using SIT Launcher to reinstall a copy of files, seemed to do the trick in getting rid of the unity error...

However, now I have the issue of it getting stuck on "Loading profile data...".

Do you know what might be the cause of it?

**Comment by w123456w30w** [comment]
may be caused by the external IP in the coopconfig.json file not matching the launcher's server IP.

**Comment by Xenoblivion53** [comment]
The external IP in the coopconfig.json is matching the IP in the launcher.

When using Hamatchi, does a port need to be forwarded as well? Cause I have not done that yet, and if I were to forward the port, would it be UDP, TCP, or both?

Thanks!

**Comment by jeremymouton** [comment]
Have you tried setting all your IPs to localhost (`127.0.0.1`), before diving in Hamachi?

`<server>/user/mods/SIT.Aki-Server-Mod/src/coopconfig.json`
```
{
    "externalIP": "http://127.0.0.1:6969",
    "webSocketPort": 6970
}
```

`<server>/Aki_Data/Server/configs/http.json`
```
{
    "ip": "127.0.0.1",
    "port": 6969,
	"webSocketPingDelayMs": 90000,
    "logRequests": true,
	"serverImagePathOverride": {	}
}
```

Then, in SIT Launch set the server to `http://127.0.0.1:6969`

I was able to get the latest version running using localhost, but it took some tinkering.

**Comment by Xenoblivion53** [comment]
Sorry for the late response, I just tried changing the IP to 127.0.0.1 and am still stuck on the "Loading profile data..." screen for quite some time. Eventually, I'm hit with a backend error, reading the following: "Backend error: Cannot connect to destination host"

I assume this has something to do with the IP / Port not working or being recognized? Are there any other potential solutions I could try?

_It is possible, that due to my lack of knowledge, I was not using Hamachi correctly and or needed to open port 6969 for this to work._

(Image of error):
![image](https://github.com/paulov-t/SIT.Core/assets/82923762/18dad368-6fea-4509-b07d-cc577b8663d8)

Thank you!

**Comment by nowblank** [comment]
@Xenoblivion53 you don't need to port forward if you are using hamachi (thats the whole point of using hamachi).. @jeremymouton also my game is stuck at loading screen using localhost 12.0.0.1:6969  ...what did you do to get past the loading screen.. using all latest release

**Comment by jeremymouton** [comment]
@nowblank I wasn't able to reproduce the fix last night. 

After retrying a fresh install, it resulted in the endless "Loading profile data..." screen again. I wasn't able to figure out what I did to fix it, but I'll give it another go this week and report back.

**Comment by SlejmUr** [comment]
Run as admin?

**Comment by jeremymouton** [comment]
I think I found the source of my issues with localhost.

The server mod does not appear to honor the `coopconfig.json` IP setting if it finds an external one to use instead.
https://github.com/paulov-t/SIT.Aki-Server-Mod/blob/master/src/mod.ts#L697-L703

It makes sense, but it also makes it difficult to troubleshoot. I think `coopconfig.json` should be the *source of truth* here. 

If ports aren't configured properly for incoming connections to your external IP, you get a `Backend error: Cannot connect to destination host` in the game logs. 

To get the game running on localhost:

- Open `<server>/user/mods/SIT.Aki-Server-Mod/src/mod.ts` (https://github.com/paulov-t/SIT.Aki-Server-Mod/blob/master/src/mod.ts#L92)
- Update the following lines:
```
this.externalIPFinder = new ExternalIPFinder();
```
to
```
this.externalIPFinder = {
    ip: 127.0.0.1,
};
```
- Relaunch server and game

**Comment by Xenoblivion53** [comment]
Updating the externalIPFinder with your new line worked like magic, and got through the infinite loading screen.

Considering this is the localhost IP we're using, I presume I just have my friend connect to my network on Hamatchi and then use the localhost, and they should connect to me?

Thank you!

**Comment by Xenoblivion53** [comment]
I went ahead and punched in my Hamachi IP in place of all the localhost IPs and it booted fine and hosted correctly. So it looks like the fix was the issue with the externalIPFinder.

I'm not quite going to close the issue yet until I can get home and make sure the server fully works by having others at least connect to it. I will be back to say how it goes.

**Comment by nowblank** [comment]
@jeremymouton @SlejmUr @Xenoblivion53 ..  local worked fine after your suggestion.. we need a way for it to work online.. since a lot of times normally due to some issues you cannot port forward (ISP issues) and thus unable to play online.... we need some sort of guide for it to work with radmin vpn or hamachi or something like that..

**Comment by SlejmUr** [comment]
Already exist 😉

**Comment by nowblank** [comment]
@SlejmUr yes I know.. but does not work now with the latest updated in coop module with the external IP finder.. I setup everything the client could not join the game only hosting option is available..

**Comment by jeremymouton** [comment]
@nowblank Some new updates were released today. Can you see if this guide helps? https://github.com/paulov-t/SIT.Core/discussions/33#discussioncomment-5995461
